### PR TITLE
ISPN-15921 Add Server Hinting unit test

### PR DESCRIPTION
**Pull request by wburns**
https://issues.redhat.com/browse/ISPN-15921

### CODE REVIEW

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/ServerHintingTest.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      hotRodServer3 = null;
+      hotRodServer4 = null;
+      remoteCacheManager = null;
+   }
+
+   @Override
+   protected void createCacheManagers() throws Throwable {
+      config = hotRodCacheConfiguration(getDefaultClusteredCacheConfig(CacheMode.DIST_SYNC, false));
+      config.clustering().hash().numOwners(1);
+      for (int i = 0; i < 3; ++i) {
```
Comment (jabolina) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/ServerHintingTest.java:72]:
```suggestion
      for (int i = 0; i < 4; ++i) {
```

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/ServerHintingTest.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      CacheTopologyInfo cti = cache2.getCacheTopologyInfo();;
+      assertEquals(3, cti.getSegmentsPerServer().entrySet().size());
+      for (Map.Entry<SocketAddress, Set<Integer>> entry : cti.getSegmentsPerServer().entrySet()) {
+         IntSet serverPrimarySegments;
+         InetSocketAddress clientAddress = (InetSocketAddress) entry.getKey();
+         if (hotRodServer1.getAddress().getPort() ==  clientAddress.getPort()) {
+            serverPrimarySegments = cache(0).getAdvancedCache().getDistributionManager().getCacheTopology().getLocalPrimarySegments();
+         } else if (hotRodServer2.getAddress().getPort() ==  clientAddress.getPort()) {
+            serverPrimarySegments = cache(1).getAdvancedCache().getDistributionManager().getCacheTopology().getLocalPrimarySegments();
+         } else {
```
Comment (jabolina) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/ServerHintingTest.java:123]: Another if for server4 here.

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/ServerHintingTest.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+   }
+
+   @Test
+   public void testClientSegmentsMapProperly() {
+
+      // Send a command just in case topology wasn't updated
+      cache2.put("foo", "bar");
+
+      CacheTopologyInfo cti = cache2.getCacheTopologyInfo();;
+      assertEquals(3, cti.getSegmentsPerServer().entrySet().size());
```
Comment (jabolina) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/ServerHintingTest.java:115]:
```suggestion
      assertEquals(4, cti.getSegmentsPerServer().entrySet().size());
```

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/ServerHintingTest.java (hunk 1 above)
Comment (wburns) [reply client/hotrod-client/src/test/java/org/infinispan/client/hotrod/ServerHintingTest.java:72]: Any reason for the extra server?

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/ServerHintingTest.java (hunk 2 above)
Comment (wburns) [reply client/hotrod-client/src/test/java/org/infinispan/client/hotrod/ServerHintingTest.java:123]: Isn't need as per previous comment

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/ServerHintingTest.java (hunk 3 above)
Comment (wburns) [reply client/hotrod-client/src/test/java/org/infinispan/client/hotrod/ServerHintingTest.java:115]: Isn't need as per previous comment

**Comment by tristantarrant** [comment]
```
22:08:36  [ERROR] /home/infinispan/workspace/Infinispan_PR-12209/client/hotrod-client/src/test/java/org/infinispan/client/hotrod/ServerHintingTest.java:19:8: Unused import - org.infinispan.commons.util.Util. [UnusedImports]
22:08:36  [ERROR] /home/infinispan/workspace/Infinispan_PR-12209/client/hotrod-client/src/test/java/org/infinispan/client/hotrod/ServerHintingTest.java:114:61: Empty statement. [EmptyStatement]
```

**Comment by tristantarrant** [comment]
I've pushed a commit which should fix the build

**Review by jabolina — APPROVED** [review]

**Comment by jabolina** [comment]
Thanks, @tristantarrant @wburns!

**Pull request merged**